src/app/Services/data.service.ts

```diff
@@ -11,4 +11,9 @@ export class DataService {
       return item.folder === box;
     });
   }
+  getEmailDetail(id: any) {
+    return data.filter((email) => {
```

Comment:
why return an array instead of an object?
use this: `return data.find((email) => email._id === id);`